int32 OUO
---
int32[] CupResult